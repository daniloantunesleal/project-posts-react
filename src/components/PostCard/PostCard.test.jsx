import { render, screen } from "@testing-library/react";
import { PostCard } from ".";

const props = {
    title: "this is the title",
    body: "this is the body",
    id: 1,
    cover: "img/daniloaleal.png",
};

describe("<PostCard />", () => {
    it("should render PostCard correctly", () => {
        render(<PostCard {...props} />);

        expect(screen.getByRole("img", { name: props.title })).toHaveAttribute("src", props.cover);
        expect(screen.getByRole("heading", { name: props.title })).toBeInTheDocument();
        expect(screen.getByText(props.title)).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        const { container } = render(<PostCard {...props} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
