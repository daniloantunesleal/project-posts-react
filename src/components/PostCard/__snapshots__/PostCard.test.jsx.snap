// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="this is the title"
    src="img/daniloaleal.png"
  />
  <div
    class="post-content"
  >
    <h2>
      this is the title
    </h2>
    <p>
      this is the body
    </p>
  </div>
</div>
`;
